src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -215,22 +216,23 @@ void QgsSymbolLegendNode::uncheckAllItems()
   checkAll( false );
 }
 
-inline
-QgsRenderContext *QgsSymbolLegendNode::createTemporaryRenderContext() const
+QgsRenderContext *QgsLayerTreeModelLegendNode::createTemporaryRenderContext() const
 {
   double scale = 0.0;
   double mupp = 0.0;
   int dpi = 0;
   if ( model() )
     model()->legendMapViewData( &mupp, &dpi, &scale );
-  bool validData = !qgsDoubleNear( mupp, 0.0 ) && dpi != 0 && !qgsDoubleNear( scale, 0.0 );
+
+  if ( qgsDoubleNear( mupp, 0.0 ) || dpi == 0 || qgsDoubleNear( scale, 0.0 ) )
+    return nullptr;
 
   // setup temporary render context
-  std::unique_ptr<QgsRenderContext> context( new QgsRenderContext );
+  QgsRenderContext *context = new QgsRenderContext;
```

Comment:
I think this should still be a unique_ptr - there should be no cost and it reduces the chance of leaks if someone changes things here in future.